File: trip-kit-booking/src/main/java/com/skedgo/android/tripkit/booking/LinkFormField.java
package com.skedgo.android.tripkit.booking;

import com.google.gson.annotations.SerializedName;

public class LinkFormField extends FormField {
  public static final String METHOD_GET = "get";
  public static final String METHOD_POST = "post";
  public static final String METHOD_REFRESH = "refresh";
  public static final String METHOD_EXTERNAL = "external";
  @SerializedName("value")
  private String link;
  @SerializedName("method")
  private String method;

  @Override
  public String getValue() {
    return link;
  }

  public void setValue(String link) {
    this.link = link;
  }

  public String getMethod() {
    return method;
  }

  public void setMethod(String method) {
    this.method = method;
  }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/LinkFormField.kt
package com.skedgo.tripkit.booking

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

class LinkFormField : FormField {

    companion object {
        const val METHOD_GET = "get"
        const val METHOD_POST = "post"
        const val METHOD_REFRESH = "refresh"
        const val METHOD_EXTERNAL = "external"

        @JvmField
        val CREATOR: Parcelable.Creator<LinkFormField> = object : Parcelable.Creator<LinkFormField> {
            override fun createFromParcel(parcel: Parcel): LinkFormField {
                parcel.readInt() // Read LINK type
                return LinkFormField(parcel)
            }

            override fun newArray(size: Int): Array<LinkFormField?> {
                return arrayOfNulls(size)
            }
        }
    }

    @SerializedName("value")
    var link: String? = null

    @SerializedName("method")
    var method: String? = null

    constructor(parcel: Parcel) : super( parcel) {  // Call the parent class constructor
        link = parcel.readString()
        method = parcel.readString()
    }

    // Secondary empty constructor (the default constructor)
    constructor() : super() {
        // Initialize any default values or leave it empty
    }

    override fun getValue(): String? {
        return link
    }

    fun setValue(link: String) {
        this.link = link
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(LINK)
        super.writeToParcel(dest, flags)
        dest.writeString(link)
        dest.writeString(method)
    }
}